pulsar-broker-common/src/main/java/org/apache/pulsar/broker/authorization/AuthorizationProvider.java

```diff
@@ -162,6 +162,24 @@
     CompletableFuture<Boolean> allowSinkOpsAsync(NamespaceName namespaceName, String role,
                                                  AuthenticationDataSource authenticationData);
 
+    /**
+     * Check the a role has the permission to do the package operations in a namespace.
+     *
+     * @param namespaceName
+     *              the namespace name you want to check
+     * @param role
+     *              the role to check
+     * @param authenticationData
+     *              authentication data related to the role
+     * @return
+     *              a boolean value to determine whether authorized or not
+     */
+    default CompletableFuture<Boolean> canDoPackageOpsAsync(NamespaceName namespaceName, String role,
```

Comment:
We need to add a method like https://github.com/apache/pulsar/pull/8893/files#diff-cea47b9766438a068de61669ccde5f4e8435941c8953edf176aa62ffddf80545R248. This is the new method of defining the permissions. Don't use the old pattern.

Reply:
It seems the link is the diff for this PR...

Reply:
https://github.com/apache/pulsar/blob/7c68ade453136895d19fa4723351d26256f3d624/pulsar-broker-common/src/main/java/org/apache/pulsar/broker/authorization/AuthorizationProvider.java#L434

Check this one.